### PR TITLE
Fall back to writing a regular file without `O_TMPFILE`

### PR DESCRIPTION
**Pull request by cgwalters**
Sadly `overlayfs` at least doesn't yet support `O_TMPFILE`
so we need the fallback path.

And this can get hit in usual cases like a build pod where
`/tmp` is actually part of the overlayfs and not a proper
host path.

This actually reveals an API problem in that the prefix/suffix
for this file isn't configurable; will file that as a
tracker for an API break.

### CODE REVIEW

**Comment by cgwalters** [comment]
CI here won't cover this, but it was easy to replicate locally with a privileged podman container which will use `overlayfs`.

File: src/lib.rs [hunk 1]
```diff
@@ -244,10 +246,48 @@ impl OpenatDirExt for openat::Dir {
         destname: P,
         mode: libc::mode_t,
     ) -> io::Result<FileWriter> {
-        let tmpf = self.new_unnamed_file(mode)?;
+        let (tmpf, name) = if let Some(tmpf) = self.new_unnamed_file(mode).ok() {
+            (tmpf, None)
+        } else {
+            // FIXME allow this to be configurable
+            let (tmpf, name) = tempfile_in(self, ".tmp", ".tmp", mode)?;
```
Comment (jlebon) [inline src/lib.rs:253]: Using `.tmp` for both the prefix and suffix seems a bit overdescriptive, but of course fine as is too.

**Review by jlebon — APPROVED** [review]
Looks sane to me overall.
/approve

File: src/lib.rs (hunk 1 above)
Comment (cgwalters) [reply src/lib.rs:253]: Yeah, just trying hard to ensure the file is ignored by default.

**Pull request merged**